storage/innobase/trx/trx0purge.cc

```diff
@@ -1549,6 +1551,17 @@ static bool trx_purge_truncate_marked_undo() {
   }
   ut_ad(mdl_ticket != nullptr);
 
+  // Acquire Percona's LOCK TABLES FOR BACKUP lock
+  if (current_thd->backup_tables_lock.is_acquired()) {
+    ib::info(ER_IB_MSG_UNDO_TRUNCATE_DELAY_BY_LTFB, space_name.c_str());
+    return (false);
+  }
+  if (current_thd->backup_tables_lock.acquire_protection(current_thd,
+                                                         MDL_TRANSACTION, 0)) {
+    ib::info(ER_IB_MSG_UNDO_TRUNCATE_DELAY_BY_LTFB, space_name.c_str());
```

Comment:
This ER_ seems new. Did you include all files for PR? :thinking:

Reply:
Good catch. Added.